docs: rewrite Quick Start as short developer guide (CTORNDGAIN-1301) (#82)
griddynamics/rosetta: docs/web/docs/quickstart.md

permalink: /docs/quickstart/
---

# Overview

Rosetta gives your AI coding agent your team's context, standards, and guardrails — across any IDE.

> You install Rosetta in your IDE, ask the agent "Initialize this repository using Rosetta", and it produces a plan + code + tests with the conventions of your codebase, not generic ones.

Reviewer: ```suggestion
You install Rosetta in your IDE, ask the agent "Initialize this repository using Rosetta", and during coding workflows it produces a plan + code + tests with the conventions of your codebase, not generic ones.
```

> > **Use a strong model.** Sonnet 4.6, GPT-5.3-codex-medium, gemini-3.1-pro, or better. Avoid Auto. Weaker models silently skip Rosetta's tools.

Reviewer: ```suggestion
> [!WARNING]
> You must receive a prior approval from your manager and company to use it.

> [!WARNING]
> Use **Sonnet 4.6**, **GPT-5.3-codex-medium**, **gemini-3.1-pro** or better models. Avoid Auto model selection.

> [!NOTE]
> Rosetta is designed to never use or see data or IP.
> Instead it uses inversion of control, by providing a "menu" to AI coding agents.
```

---

## 1. Install

Two options. **Pick A if your IDE supports a plugin** — it's the recommended path. Otherwise use B.

### Option A — Plugin (recommended)

> A plugin bundles Rosetta's bootstrap rule, skills, agents, and workflows directly into your IDE. No MCP wiring, no manual bootstrap file.

Reviewer: ```suggestion
A plugin bundles Rosetta's bootstrap rule, skills, agents, and workflows directly into your IDE. No MCP wiring, no manual bootstrap file.

More details: /rosetta/docs/installation/#plugin-based-installation
```

| IDE                          | Install                                                                                                                                    |
| ---------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------ |
| **Claude Code**              | `claude plugin marketplace add griddynamics/rosetta` then `claude plugin install core@rosetta`                                              |
| **VS Code / GitHub Copilot** | Install `core-copilot` via VS Code Copilot Plugins                                                                                          |

> | **JetBrains / Copilot**      | Zip + manual config — see [Installation](/rosetta/docs/installation/#plugin-based-installation)                                             |

Reviewer: ```suggestion
| **JetBrains / Copilot**      | Unpack Zip + Enable Agentic Features                      |
```

> | **Codex**                    | Zip + `codex features enable codex_hooks` — see [Installation](/rosetta/docs/installation/#plugin-based-installation)                       |

Reviewer: ```suggestion
| **Codex**                    | Unpack Zip + `codex features enable hooks`                |
```

| Cursor                         | `.cursor/rules/bootstrap.mdc`     |
| Claude Code                    | `.claude/claude.md`               |
| VS Code / Copilot (any)        | `.github/copilot-instructions.md` |
| JetBrains Junie                | `.junie/guidelines.md`            |
| Windsurf                       | `.windsurf/rules/bootstrap.md`    |
| Antigravity                    | `.agent/rules/bootstrap.md`       |
| OpenCode                       | `AGENTS.md`                       |

Keep the file's YAML frontmatter intact.

## 2. Verify

```
What can you do, Rosetta?
```

The agent should list Rosetta's workflows and capabilities. If it doesn't, jump to [Setup issues](#setup-issues).

## 3. Initialize your repo

```
Initialize this repository using Rosetta
```

Rosetta scans your stack, generates `docs/TECHSTACK.md`, `docs/CODEMAP.md`, `docs/DEPENDENCIES.md`, `docs/CONTEXT.md`, `docs/ARCHITECTURE.md`, and asks you to fill the gaps. **Restart your chat session afterward** so the new context loads.

Run once per repo. For composite workspaces, init each repo first, then run init at workspace level with "This is composite workspace" appended.

---

> ## A real example

Reviewer: Check PLUGINS next steps and put them in QuickStart. Do not allow AI hallucinations.
